pkg/cosign/rego/rego.go

```diff
@@ -93,27 +100,57 @@ func ValidateJSONWithModuleInput(jsonBody []byte, moduleInput string) error {
 
 	evalQuery, err := r.PrepareForEval(ctx)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	var input interface{}
 	dec := json.NewDecoder(bytes.NewBuffer(jsonBody))
 	dec.UseNumber()
 	if err := dec.Decode(&input); err != nil {
-		return err
+		return nil, err
 	}
 
 	rs, err := evalQuery.Eval(ctx, rego.EvalInput(input))
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	for _, result := range rs {
-		isCompliant, ok := result.Bindings[CosignEvaluationRule].(bool)
-		if ok && isCompliant {
-			return nil
+		switch response := result.Bindings[CosignEvaluationRule].(type) {
+		case []interface{}:
+			return evaluateRegoEvalMapResult(query, response)
+		case bool:
+			if response {
+				return nil, nil
+			}
 		}
 	}
 
-	return fmt.Errorf("policy is not compliant for query '%s'", query)
+	return nil, fmt.Errorf("policy is not compliant for query '%s'", query)
+}
+
+func evaluateRegoEvalMapResult(query string, response []interface{}) (warnings error, errors error) {
+	var warnMsg error
+	errMsg := fmt.Errorf("policy is not compliant for query '%s'", query)
```

Comment:
nit: here and below you can use `%q` to quote things.

Reply:
Done